app/lib/otis/registration_mover.rb

```diff
@@ -16,6 +10,7 @@ def ht_user
       return @ht_user unless @ht_user.nil?
 
       institution = HTInstitution.find(@registration.inst_id)
+      @ht_user = @registration.existing_user || HTUser.new
```

Comment:
Does this `@ht_user` not immediately get overwritten in the next line?

Reply:
@moseshll when I create a new registration with an existing email address, I get the "existing user" badge, but when I try to finish the registration I get a warning about "E-mail has already been taken".

Reply:
That's bonkers. It looks like the thing got munged when I rebased, because there was quite a bit of clowning around that had to be done in the registration mover -- Rails gets irked when you try to change a primary key and just discards the new value.

Clearly the testing is faulty as well, because this never should have gotten by.

Reply:
Fixed logic and added some sanity checks to tests.

Reply:
@respinos Update deployed to testing. Is your testing data in a state to try a resubmit?

Reply:
@moseshll Now I get `User ID can't be blank and Multi-Factor must be blank` --- but if it was attaching to the existing `ht_user` then there'd be a username, right?

Reply:
Those are heavily dependent on the Shib `otis_registrations.env` picked up by the "finalize" process. Leaving aside uniqname weirdness, user ID will be from HTTP_X_REMOTE_USER and it would overwrite any existing value for `ht_users.userid`. The multi-factor error is a bit more concerning. I take it you are running this locally? That bit should be taking care of itself so long as the institution the new user is attaching to is well-formed.